asmcli/commands/create-mesh.sh

```diff
@@ -70,105 +74,94 @@ create-mesh_validate_args() {
-  # validate clusters are valid
+  # generate kubeconfig files for each cluster P/L/C
   while read -r PROJECT_ID CLUSTER_LOCATION CLUSTER_NAME; do
-    validate_cluster "${PROJECT_ID}" "${CLUSTER_LOCATION}" "${CLUSTER_NAME}"
-    configure_kubectl "${PROJECT_ID}" "${CLUSTER_LOCATION}" "${CLUSTER_NAME}"
+    # set KCF to the new kubeconfig
+    KCF="$(mktemp)"
 
-    CTX_CLUSTER="$(kubectl config current-context)"
-    if ! is_membership_crd_installed; then
-      GKE_CLUSTER_URI="$(retry 2 gcloud container clusters describe "${CLUSTER_NAME}" \
-      --zone="${CLUSTER_LOCATION}" \
+    # generate kubeconfig
+    info "Fetching/writing GCP credentials to ${KCF}..."
+    KUBECONFIG="${KCF}" retry 2 gcloud container clusters get-credentials "${CLUSTER_NAME}" \
       --project="${PROJECT_ID}" \
-      --format='value(selfLink)')"
-      context_append "clusterRegistrations" "${CTX_CLUSTER} ${GKE_CLUSTER_URI}"
-    else
-      exit_if_cluster_registered_to_another_fleet "${PROJECT_ID}" "${CLUSTER_LOCATION}" "${CLUSTER_NAME}"
-      warn "Cluster ${CLUSTER_NAME} is already registered with fleet ${FLEET_ID}. Skipping."
-    fi
-    context_append "clusterContexts" "${CTX_CLUSTER}"
+      --zone="${CLUSTER_LOCATION}"
+    
+    # save the kubeconfig to context
+    context_append "kubeconfigFiles" "${KCF}"
   done <<EOF
 $(context_list "clustersInfo")
 EOF
-}
 
-create_mesh() {
-  local CTX_CLUSTER GKE_CLUSTER_URI
+  # required because registration command is diffv2com  erent for GKE vs off-GKE
+  create-mesh_set_platform
 
-  # for-loop does not read lines but words, so setting IFS to explicitly split with line breaks
-  # Also context_list might return an empty list so we use for-loop to bypass that scenario
-  while read -r CTX_CLUSTER GKE_CLUSTER_URI; do
-    add_one_to_mesh "${CTX_CLUSTER}" "${GKE_CLUSTER_URI}"
-  done < <(context_list "clusterRegistrations")
+  # validate clusters are valid
+  while read -r KCF; do
+    context_set-option "KUBECONFIG" "${KCF}"
+    context_set-option "CONTEXT" "$(kubectl config current-context)"
+    is_cluster_registered
+  done <<EOF
+$(context_list "kubeconfigFiles")
+EOF
 }
 
-add_one_to_mesh() {
-  local CTX_CLUSTER; CTX_CLUSTER="${1}"
-  local GKE_CLUSTER_URI; GKE_CLUSTER_URI="${2}"
-  local PROJECT_ID CLUSTER_LOCATION CLUSTER_NAME MEMBERSHIP_NAME
-  IFS='_' read -r _ PROJECT_ID CLUSTER_LOCATION CLUSTER_NAME < <(echo "$CTX_CLUSTER")
-  local FLEET_ID; FLEET_ID="$(context_get-option "FLEET_ID")"
-
-  MEMBERSHIP_NAME="$(generate_membership_name "${PROJECT_ID}" "${CLUSTER_LOCATION}" "${CLUSTER_NAME}")"
-
-  info "Registering the cluster ${PROJECT_ID}/${CLUSTER_LOCATION}/${CLUSTER_NAME} as ${MEMBERSHIP_NAME}..."
+create-mesh_set_platform() {
+  while read -r KCF; do
+    context_set-option "KUBECONFIG" "${KCF}"
+    if [[ "$(kubectl config current-context)" =~ "gke_" ]]; then
+      context_set-option "PLATFORM" "gcp"
+    else
+      context_set-option "PLATFORM" "multicloud"
+    fi
+    break
+  done <<EOF
```

Comment:
If the intention here is to just get the first line, you can do it with head.
```
KFC="$(context_list "kubeconfigFiles" | head -n 1)"
if [[ -z "${KFC}" ]]; then fatal "Context is empty? Please file a GitHub issue."; fi
context_set-option "KUBECONFIG" ....
```

Reply:
Yes, thanks!